### PR TITLE
OUTPOST-64 bulk import services

### PR DESCRIPTION
**Pull request by apricot13**
- [x] documentation
- [x] import services with all core fields
- [x] import custom fields
- [x] tests

### CODE REVIEW

File: lib/tasks/data_import/services/output_service_delimited_fields.rake [hunk 1]
```diff
@@ -0,0 +1,55 @@
+namespace :services do
+    desc 'Output a list of all the data contained in the delimited fields'
+    task :display_delimited_field_values => :environment do
+      file_path = Rails.root.join('lib', 'seeds', 'tvp-services.csv')
```
Comment (em-cd) [inline lib/tasks/data_import/services/output_service_delimited_fields.rake:4]: Sorry if this is a silly question 😄 Is this rake task a kind of verification step to see what the data will look like once it's imported, like a dry run?

File: lib/tasks/data_import/services/README.md [hunk 2]
```diff
@@ -28,62 +28,62 @@ In order to add multiple locations to a service you would create one entry for y
...
+| `information_sharing`                                                                                                           | Text  |  How will you share information with me about my child’s progress? What is additional for children with SEND? |  |  
+| `environment_accessibility`                                                                                                     | Text | How accessible is the environment (indoors and outdoors)?  |   |  
+| `how_to_start`                                                                                                                  | Text  |  How can children and young people with SEND and their families start to use your service or activity? |  |  
+| `future_plans`                                                                                                                  | Text  | What future plans do you have for developing your SEND provision?  |  |  
+| `custom_<custom_field_type>_field_name`                                                                                         | depends  | Format for custom fields  |  |
```
Comment (em-cd) [inline lib/tasks/data_import/services/README.md:86]: I don't understand how this is meant to be used - how do we know which custom field we're adding service meta for - for example if there is more than one `text` type custom field?

File: lib/tasks/data_import/services/output_service_delimited_fields.rake (hunk 1 above)
Comment (apricot13) [reply lib/tasks/data_import/services/output_service_delimited_fields.rake:4]: yeh - a sanity check, I've done a similar function in google but its a bit flakey. 

I can see this being useful later on as a kind of 'preview your import' page as well

File: lib/tasks/data_import/services/README.md (hunk 2 above)
Comment (apricot13) [reply lib/tasks/data_import/services/README.md:86]: those fields are for the SEND needs section.

Custom fields is this bit
`custom_<custom_field_type>_field_name | depends | Format for custom fields
-- | -- | --
`



if we had a custom field called "Type of guinea pig" that was free text (created previously with the custom import script or already in the site) then the column name would be `custom_text_type_of_guinea_pig` if we had one that was a select field for "What is the best vegetable for piggies?" there would be a column `column_select_what_is_the_best_vegetable_for_piggies` with content like "cucumbers;red peppers"

File: lib/tasks/data_import/services/output_service_delimited_fields.rake (hunk 1 above)
Comment (apricot13) [reply lib/tasks/data_import/services/output_service_delimited_fields.rake:4]: it also works as an additional form of validation!

**Comment by apricot13** [comment]
I cant 👍  it but 👍 !

File: lib/tasks/data_import/services/output_service_delimited_fields.rake (hunk 1 above)
Comment (em-cd) [reply lib/tasks/data_import/services/output_service_delimited_fields.rake]: Cool, thanks! I've added a bit to the README about it.

File: lib/tasks/data_import/services/README.md (hunk 2 above)
Comment (em-cd) [reply lib/tasks/data_import/services/README.md]: This makes sense to me now! I've added an example to the README.

File: lib/tasks/data_import/services/import_services.rake [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+          new_service_send_needs(new_service, row['send_needs_support'])
+
+
+          # custom fields
+
+          #yuck
+          # @TODO loop through CustomField.types here
+          text_rows = row.headers.filter{|h|h.starts_with?('custom_text_')}.map{|e| e.slice('custom_text_'.length, e.length) }
+          text_field_info = []
+          text_field = CustomField.where(field_type: 'text').map{|m| text_field_info << {"id" => m['id'], "key" => m['key'], "field" => m['key'].downcase.delete("^a-zA-Z0-9 ").gsub(' ', '_')}}
```
Comment (em-cd) [inline lib/tasks/data_import/services/import_services.rake:169]: We don't seem to be using this `text_field` variable anywhere, or any of the similar ones for the other custom field types below. Do we need them?

**Comment by apricot13** [comment]
a work of art! now this song is stuck in my head! 😂 
https://www.youtube.com/watch?v=gXk0QDcRwkU

**Pull request merged**